File: Java codes/QueueUsingLinkedList.java
import java.io.*;

class Node
{
	int data;
	Node next;
	Node(int data)
	{
		this.data = data;
	}
}

class Queue
{
	private Node front;
	private Node rear;

	Queue()
	{
		this.front = this.rear = null;
	}

	public void enqueue(int val)
	{
		if (front == null && rear == null)
		{
			rear = new Node(val);
			front = rear;
		}
		else
		{
			rear.next = new Node(val);
			rear = rear.next;
		}
	}

	public int dequeue()
	{
		int val = front.data;
		if (front == rear)
			front = rear = null;
		else
			front = front.next;
		return val;
	}

	public int checkFront()
	{
		return front.data;
	}

	public boolean isEmpty()
	{
		if (front == null && rear == null)
			return true;
		return false;
	}
}

class QueueUsingLinkedList
{
	public static void main(String[] args) throws IOException
	{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		Queue queue = new Queue();
		while (true)
		{
			System.out.println("1- Enqueue\n2- Dequeue\n3- Check Front\n4- Check if empty\n5- Quit");
			int choice = Integer.parseInt(br.readLine());
			switch(choice)
			{
				case 1:
					System.out.println("Value- ");
					queue.enqueue(Integer.parseInt(br.readLine()));
					break;

				case 2:
					if (queue.isEmpty())
						System.out.println("Empty Queue");
					else
						System.out.println(queue.dequeue()+" removed");
					break;

				case 3:
					if (queue.isEmpty())
						System.out.println("Empty Queue");
					else
						System.out.println(queue.checkFront()+" is at the front");
					break;

				case 4:
					if (queue.isEmpty())
						System.out.println("Empty");
					else
						System.out.println("Not empty");
					break;

				case 5:
					System.out.println("Quit");
					System.exit(0);

				default:
					System.out.println("Invalid Input");
			}
		}
	}
}

File: C codes/QueueUsingLinkedList.c
#include <stdio.h>
#include <stdlib.h>

struct Queue
{
	int num;
	struct Queue *next;
};

void enqueue();
void dequeue();
void display();

struct Queue *front=NULL, *rear=NULL, *node, *temp;

void enqueue()
{
		temp=(struct Queue *)malloc(sizeof(struct Queue));
		printf("Enter the information-\n");
		scanf("%d",&temp->num);
		if(front==NULL&&rear==NULL)
		{
			temp->next=NULL;
			front=temp;
			rear=temp;
		}
		else
		{
			temp->next=NULL;
			rear->next=temp;
			rear=rear->next;
		}
		display();
}

void dequeue()
{
	if(front==NULL&&rear==NULL)
		printf("Queue underflow\n");
	else if(front==rear)
	{
		printf("%d is deleted\n",front->num);
		temp=front;
		front=NULL;
		rear=NULL;
		free(temp);
		display();
	}
	else
	{
		printf("%d is deleted\n",front->num);
		temp=front;
		front=front->next;
		temp->next=NULL;
		free(temp);
		display();
	}
}

void display()
{
	if(front==NULL&&rear==NULL)
		printf("Queue is empty\n");
	else
	{
		printf("The queue is-\n");
		node=front;
		while(node!=NULL)
		{
			printf("%d\t",node->num);
			node=node->next;
		}
	}
}

void main()
{
	printf("\t\t\tQUEUE\n");
	while(1)
	{
		int choice;
		printf("\n\t\t\tSEE THE OPTIONS\n\t\t\t1- Enqueue\n\t\t\t2- Dequeue\n\t\t\t3- Display\n\t\t\t4- Quit\n\t\t\tEnter your choice-\n");
		scanf("%d",&choice);
		switch(choice)
		{
			case 1:
				enqueue();
				break;
			case 2:
				dequeue();
				break;
			case 3:
				display();
				break;
			case 4:
				printf("\t\t\tTHANKS FOR USING MY APPLICATION!!!");
				return;
				break;
			default:
				printf("INVALID INPUT\n");
				break;
		}
	}
}